modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/dnd_table/Row.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, {useContext, useMemo} from 'react';
+
+import Cell from './Cell';
+import TableContext from './TableContext';
+
+function Row({children, className, paddingLeftCells}) {
+	const {columnsDefinitions, columnsNames, isFixed} = useContext(
+		TableContext
+	);
+
+	const marginLeft = useMemo(() => {
+		let margin = 0;
+
+		if (isFixed) {
+			for (let i = 0; i < paddingLeftCells; i++) {
+				margin += columnsDefinitions.get(columnsNames[i]).width;
+			}
+		}
+
+		return margin;
+	}, [columnsDefinitions, columnsNames, isFixed, paddingLeftCells]);
+
+	const style = marginLeft
+		? {
+				marginLeft,
+				minWidth: `calc(100% - ${marginLeft}px)`,
+		  }
+		: {};
+
+	let counter = 0;
+	const placeholderPaddingCells = !isFixed
+		? new Array(paddingLeftCells).fill(<Cell key={counter++} />)
```

Comment:
Are you sure this does what you want? `counter++` is going to be incremented exactly once here, and the  same instance of the `<Cell />` component is going to appear in every slot of the array.

"Proof":

```js
let counter = 0;

const array = new Array(2).fill({key: counter++});

console.log(array[0]); // {key: 0}
console.log(array[1]); // {key: 0}
console.log(array[0] === array[1]); // true
```

I suspect you want something like:

```js
const placeholderPaddingCells = [...Array(paddingLeftCells).keys()].map((key) => <Cell key={key} />);
```

Seeing as clever is the new dumb, this is probably better:

```js
const placeholderPaddingCells = [];

for (let i = 0; i < paddingLeftCells; i++) {
    placeholderPaddingCells.push(<Cell key={i} />);
}
```